components/kubeflow-pipeline/pipeline_components/analytics-server/flwr_analytics_server/fedbox.py

```diff
@@ -0,0 +1,102 @@
+import io
+import json
+from argparse import ArgumentParser, Namespace
+from typing import Dict, List, Tuple
+from .fivenum import fiveNumSummary
+from .fedhist import aggregate_histograms, HistogramData
+import matplotlib.pyplot as plt
+import numpy as np
+from flwr.common import Properties, Scalar
+from .provider import AnalyticsProvider, scalar_to_numpy
+
+
+class BoxPlotProvider(AnalyticsProvider):
+    def __init__(self, hrange:Tuple) -> None:
```

Comment:
Nit: Let's remove the `hrange` parameter here, as it'll be set in `set_parameters`. This way we will have a defined flow of events that set these values.

Reply:
Added this into `set_arguments`.